docs/getting_started/execution-modes.rst

```diff
@@ -238,6 +243,40 @@ Each task will create a new Cloud Run Job execution, giving full isolation. The
         },
     )
 
+Airflow Async
+-------------
+
+.. versionadded:: 1.7.0
+
+(**Experimental**)
+
+The ``airflow_async`` execution mode is a way to run the dbt resources from your dbt project using Apache Airflow's
+`Deferrable operators <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html>`__.
+This execution mode could be preferred when you've long running resources and you want to run them asynchronously by
+leveraging Airflow's deferrable operators. With that, you would be able to potentially observe higher throughput of tasks
+as more dbt nodes will be run in parallel since they won't be blocking Airflow's worker slots.
+
+In this mode, Cosmos adds a new operator, ``DbtCompileAirflowAsyncOperator``, as a root task in the DAG. The task runs
+the ``dbt compile`` command on your dbt project which then outputs compiled SQLs in the project's target directory.
+As part of the same task run, these compiled SQLs are then stored remotely to a remote path set using the
+:ref:`remote_target_path` configuration. The remote path is then used by the subsequent tasks in the DAG to
+fetch (from the remote path) and run the compiled SQLs asynchronously using e.g. the ``DbtRunAirflowAsyncOperator``.
+You may observe that the compile task takes a bit longer to run due to the latency of storing the compiled SQLs remotely,
+however, it is still a win as it is one-time overhead and the subsequent tasks run asynchronously utilising the Airflow's
+deferrable operators and supplying to them those compiled SQLs.
+
+Note that currently, the ``airflow_async`` execution mode has the following limitations and is released as Experimental:
+
+1. Only supports the ``dbt resource type`` models to be run asynchronously using Airflow deferrable operators. All other resources are executed synchronously using dbt commands as they are in the ``local`` execution mode.
+2. Only supports BigQuery as the target database. If a profile target other than BigQuery is specified, Cosmos will error out saying that the target database is not supported with this execution mode.
```

Comment:
3. Only works for `full_refresh` models. There is pending work to support other modes.

Reply:
added so.